src/Model/ShopifyFile.php

```diff
@@ -162,24 +252,118 @@ public static function getByShopifyID($shopifyId)
     }
 
     /**
-     * Download the image from the shopify CDN
-     *
-     * todo - create method in ShopifyClient for $request, update function for graphql
-     *
-     * @param $src
-     * @param $folder
-     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @param string|null $format
+     * @param int $height
+     * @return ShopifyFileSource|null
+     */
+    public function getFormat($format = null, $height = 0)
+    {
+        if ($format === null) {
+            return $this->OriginalSource();
+        }
+        $filter = [
+            'Format' => $format,
+        ];
+
+        if ($height) {
+            $filter['Height'] = $height;
+        }
+
+        return $this->Sources()->filter($filter)->first();
+    }
+
+    /**
+     * @param int|string $width
+     * @param int|string $height
+     * @param string $format
+     * @return string
+     */
+    private function generateTransformedURL($width = '', $height = '', $format = '')
+    {
+        $pattern = '/(.*)(\.\S+?)(\?.*)/m';
+        $dims = sprintf('_%sx%s', $width, $height);
+        if ($format !== 'png' && $format !== 'jpg' && $format !== 'webm') {
+            $format = '';
+        }
+
+        preg_match_all('/.*\.(\S+?)\?.*/m', $this->OriginalSource()->URL, $matches);
+        if (count($matches) > 1) {
+            if ($matches[1][0] === $format) {
+                $format = '';
+            }
+        }
+        if ($format !== '') {
+            $format = '.' . $format;
+        }
+
+        $subst = '$1' . $dims . '$2' . $format . '$3';
+        return preg_replace($pattern, $subst, $this->OriginalSource()->URL);
+    }
+
+    /**
+     * @param int $width
+     * @param int $height
+     * @param string $format
+     * @return ShopifyFileSource|null
+     */
+    public function getTransform($width, $height, $format = null)
+    {
+        if ($this->Type === 'EXTERNAL_VIDEO') {
```

Comment:
Just noting it on this one, but do we want to do things like `const EXTERNAL_VIDEO = 'EXTERNAL_VIDEO'`, then call `static::EXTERNAL_VIDEO`? Versioned does something similar likely to reduce the possibility of typos downstream if others extend the functionality.